x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/gcp_credential_form.tsx

```diff
@@ -0,0 +1,224 @@
+        defaultMessage: 'Project ID',
+      }),
+    },
+    credentials_file: {
+      label: i18n.translate('xpack.csp.gcpIntegration.credentialsLabel', {
+        defaultMessage: 'Credentials File',
+      }),
+    },
+    credentials_json: {
+      label: i18n.translate('xpack.csp.gcpIntegration.credentialsjsonLabel', {
+        defaultMessage: 'Credentials JSON',
+      }),
+    },
+  },
+};
+
+const jsonOptionsHolder = [
+  { label: 'Option A', text: 'Option A' },
+  { label: 'Option B', text: 'Option B' },
+];
+
+const getSetupFormatOptions = (): Array<{
+  id: SetupFormatGCP;
+  label: string;
+  disabled: boolean;
+}> => [
+  {
+    id: 'google_cloud_shell',
+    label: 'Google Cloud Shell',
+    disabled: true,
+  },
+  {
+    id: 'manual',
+    label: i18n.translate('xpack.csp.gcpIntegration.setupFormatOptions.manual', {
+      defaultMessage: 'Manual',
+    }),
+    disabled: false,
+  },
+];
+
+export type AwsCredentialsType = keyof typeof gcpField;
+
+interface Props {
+  newPolicy: NewPackagePolicy;
+  input: Extract<
+    NewPackagePolicyPostureInput,
+    { type: 'cloudbeat/cis_aws' | 'cloudbeat/cis_eks' | 'cloudbeat/cis_gcp' }
+  >;
+  updatePolicy(updatedPolicy: NewPackagePolicy): void;
+}
+
+const getInputVarsFields = (input: NewPackagePolicyInput, fields: GcpFields[keyof GcpFields]) =>
+  Object.entries(input.streams[0].vars || {})
+    .filter(([id]) => id in fields)
+    .map(([id, inputVar]) => {
+      const field = fields[id];
+      return {
+        id,
+        label: field.label,
+        type: field.type || 'text',
+        value: inputVar.value,
+      } as const;
+    });
+
+export const GcpCredentialsForm = ({ input, newPolicy, updatePolicy }: Props) => {
+  // We only have a value for 'aws.credentials.type' once the form has mounted.
+  // On initial render we don't have that value so we default to the first option.
+  // const awsCredentialsType = getAwsCredentialsType(input) || AWS_CREDENTIALS_OPTIONS[0].id;
+  //   const group = gcpField;
+  const fields = getInputVarsFields(input, gcpField.fields);
+
+  return (
+    <>
+      <GCPSetupInfoContent />
+      <EuiSpacer size="l" />
+      <GcpSetupAccessSelector
+        onChange={(optionId) =>
+          updatePolicy(
+            getPosturePolicy(newPolicy, input.type, {
+              'aws.credentials.type': { value: optionId },
+            })
+          )
+        }
+      />
+      <EuiSpacer size="l" />
+      <GcpInputVarFields
+        fields={fields}
+        onChange={(key, value) =>
+          updatePolicy(getPosturePolicy(newPolicy, input.type, { [key]: { value } }))
+        }
+      />
+      <EuiSpacer size="s" />
+      {DocsLink}
+      <EuiSpacer />
+    </>
+  );
+};
+
+const GcpSetupAccessSelector = ({ onChange }: { onChange(type: AwsCredentialsType): void }) => (
+  <RadioGroup
+    size="s"
+    options={getSetupFormatOptions()}
+    idSelected={'manual'}
+    onChange={(id) => onChange(id as AwsCredentialsType)}
```

Comment:
```suggestion
    onChange={(id: AwsCredentialsType) => onChange(id)}
```
This would throw type errors if the passed value do not match, while the previous way just takes any value and we "lie" to it that it's matching the type